src/components/shared/TenantSelector.tsx

```diff
@@ -0,0 +1,96 @@
+import { Skeleton } from '@mui/material';
+import AutocompletedField from 'components/shared/toolbar/AutocompletedField';
+import { useTenantDetails } from 'context/fetcher/Tenant';
+import useGlobalSearchParams, {
+    GlobalSearchParams,
+} from 'hooks/searchParams/useGlobalSearchParams';
+import { noop } from 'lodash';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { useTenantStore } from 'stores/Tenant/Store';
+import { hasLength } from 'utils/misc-utils';
+
+interface Props {
+    updateStoreState?: (value?: string) => void;
+}
+
+function TenantSelector({ updateStoreState }: Props) {
+    const intl = useIntl();
+
+    const tenants = useTenantDetails();
+    const selectedTenant = useTenantStore((state) => state.selectedTenant);
+    const setSelectedTenant = useTenantStore(
+        (state) => state.updateSelectedTenant
+    );
+
+    const handledDefault = useRef(false);
+    const defaultSelectedTenant = useGlobalSearchParams(
+        GlobalSearchParams.PREFIX
+    );
+    const [inputValue, setInputValue] = useState('');
+    const [defaultValue, setDefaultValue] = useState<string | null>(null);
+
+    const tenantNames = useMemo<string[] | null>(
+        () =>
+            tenants && tenants.length > 0
+                ? tenants.map(({ tenant }) => tenant)
+                : null,
+        [tenants]
+    );
```

Comment:
No change needed

I get why we're doing this... but I dislike it for some reason. I wonder if we should move this kinda thing into the context as a stand alone hook.

Reply:
I would argue that it should be moved into the `useTenantDetails` hook defined inside the context provider because almost every component that uses this hook has a similar, memoized check.

Reply:
Up to you if you wanna mark with a `todo` or include in this PR.

Reply:
Noting this action item in a `TODO` comment in the tenant context provider.